include/deal.II/lac/block_vector_base.h

```diff
@@ -2264,6 +2262,17 @@ void BlockVectorBase<VectorType>::extract_subvector_to (ForwardIterator
     }
 }
 
+
+template <typename VectorType>
+inline
+bool BlockVectorBase<VectorType>::supports_distributed_data() const
+{
+  // Create a dummy vector of type VectorType to check
+  // supports_distributed_data even if BlockVectorBase contains no block
+  VectorType dummy;
+  return dummy.supports_distributed_data();
```

Comment:
This makes no sense to me. The way you wrote this function, it doesn't depend on the state of the block vector -- the function may as well be `static`. I think what you probably want is to loop over all blocks and combine what *they* report for this function in some useful way. If there are no blocks, you might as well throw an exception.